api/v1alpha1/fenceagentsremediation_webhook.go

```diff
@@ -52,13 +54,19 @@ var _ webhook.Validator = &FenceAgentsRemediation{}
 // ValidateCreate implements webhook.Validator so a webhook will be registered for the type
 func (far *FenceAgentsRemediation) ValidateCreate() (admission.Warnings, error) {
 	webhookFARLog.Info("validate create", "name", far.Name)
-	return validateAgentName(far.Spec.Agent)
+	if _, err := validateAgentName(far.Spec.Agent); err != nil {
+		return nil, err
+	}
+	return validateStrategy(far.Spec.RemediationStrategy)
 }
 
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
 func (far *FenceAgentsRemediation) ValidateUpdate(old runtime.Object) (admission.Warnings, error) {
 	webhookFARLog.Info("validate update", "name", far.Name)
-	return validateAgentName(far.Spec.Agent)
+	if _, err := validateAgentName(far.Spec.Agent); err != nil {
```

Comment:
since both Create and Update run the same validations, what about extracting both checks into a generic validation function?

Reply:
I will change both validations to a generic one.

Reply:
Changed to the generic one.